persistence/remote/src/main/java/org/infinispan/persistence/remote/upgrade/HotRodTargetMigrator.java

```diff
@@ -109,42 +106,34 @@ public long synchronizeData(Cache<Object, Object> cache, int readBatch, int thre
             throw log.missingMigrationData(cache.getName());
 
          } else {
-
-            CacheTopologyInfo cacheTopologyInfo = storeCache.getCacheTopologyInfo();
-            Map<SocketAddress, Set<Integer>> remoteServers = cacheTopologyInfo.getSegmentsPerServer();
-            if (remoteServers.size() == 1) {
-               final AtomicInteger count = new AtomicInteger(0);
-               ExecutorService es = Executors.newFixedThreadPool(threads);
-               migrateEntriesWithMetadata(storeCache, cache, es, knownKeys, count, null, readBatch);
-               gracefulShutdown(es);
-               return count.get();
-            } else {
-               int numSegments = cacheTopologyInfo.getNumSegments();
-               List<Address> servers = cache.getAdvancedCache().getDistributionManager().getWriteConsistentHash().getMembers();
-               List<List<Integer>> partitions = split(range(numSegments), servers.size());
-               DistributedExecutorService executor = new DefaultExecutorService(cache);
-               Iterator<Address> iterator = servers.iterator();
-               List<CompletableFuture<Integer>> futures = new ArrayList<>(servers.size());
-               for (List<Integer> partition : partitions) {
-                  Set<Integer> segmentSet = new HashSet<>();
-                  segmentSet.addAll(partition);
-                  DistributedTask<Integer> task = executor
+            CacheTopologyInfo sourceCacheTopologyInfo = remoteSourceCache.getCacheTopologyInfo();
+            int sourceSegments = sourceCacheTopologyInfo.getNumSegments();
+            List<Address> targetServers = cache.getAdvancedCache().getDistributionManager().getWriteConsistentHash().getMembers();
+
+            List<List<Integer>> partitions = split(range(sourceSegments), targetServers.size());
+            DistributedExecutorService executor = new DefaultExecutorService(cache);
```

Comment:
We need to shut this down after we are done with it, else a thread can be leaked.